app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java

```diff
@@ -0,0 +1,218 @@
+        View rootview = inflater.inflate(R.layout.fragment_excelto_pdf, container,
+                false);
+        mSharedPreferences = PreferenceManager.getDefaultSharedPreferences(mActivity);
+        mMorphButtonUtility = new MorphButtonUtility(mActivity);
+        ButterKnife.bind(this, rootview);
+        mMorphButtonUtility.morphToGrey(mCreateExcelPdf, mMorphButtonUtility.integer());
+        mCreateExcelPdf.setEnabled(false);
+        return rootview;
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        mActivity = (Activity) context;
+        mFileUtils = new FileUtils(mActivity);
+    }
+
+    @OnClick(R.id.select_excel_file)
+    public void selectExcelFile() {
+        if (mButtonClicked == 0) {
+            Uri uri = Uri.parse(Environment.getRootDirectory() + "/");
+            Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+            intent.setDataAndType(uri, "*/*");
+            intent.addCategory(Intent.CATEGORY_OPENABLE);
+            try {
+                startActivityForResult(
+                        Intent.createChooser(intent, String.valueOf(R.string.select_file)),
+                        mFileSelectCode);
+            } catch (android.content.ActivityNotFoundException ex) {
+                showSnackbar(mActivity, R.string.install_file_manager);
+            }
+            mButtonClicked = 1;
+        }
+    }
+
+    @OnClick(R.id.create_excel_to_pdf)
+    public void openExcelToPdf() {
+        if (!mPermissionGranted) {
+            getRuntimePermissions();
+            return;
+        }
+        new MaterialDialog.Builder(mActivity)
+                .title(R.string.creating_pdf)
+                .content(R.string.enter_file_name)
+                .input(getString(R.string.example), null, (dialog, input) -> {
+                    if (StringUtils.isEmpty(input)) {
+                        showSnackbar(mActivity, R.string.snackbar_name_not_blank);
+                    } else {
+                        final String inputName = input.toString();
+                        if (!mFileUtils.isFileExist(inputName + getString(R.string.pdf_ext))) {
+                            convertToPdf(inputName);
+                        } else {
+                            MaterialDialog.Builder builder = createOverwriteDialog(mActivity);
+                            builder.onPositive((dialog12, which) -> convertToPdf(inputName))
+                                    .onNegative((dialog1, which) -> openExcelToPdf())
+                                    .show();
+                        }
+                    }
+                })
+                .show();
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        mButtonClicked = 0;
+        switch (requestCode) {
+            case mFileSelectCode:
+                if (resultCode == RESULT_OK) {
+                    mExcelFileUri = data.getData();
```

Comment:
Instead of uri, try this using this to get file path
https://github.com/Swati4star/Images-to-PDF/blob/7ef993aa2dac8628d5162fcd10f8c58d75d0ccac/app/src/main/java/swati4star/createpdf/fragment/ZipToPdfFragment.java#L92

Try and see if this fixes the issue

Reply:
@Swati4star I have tried that in this push. That error seems to go away now.